config/cli.modules.d/foreman_admin_logging_katello.yml

```diff
@@ -58,6 +58,22 @@
             :action: ensure_line_is_present
             :line: ["log_level", ": ", "INFO"]
             :join_with: ''
+      -
+        :name: pulpcore
+        :friendly_name: Pulpcore
+        :file: /etc/pulp/settings.py
+        :destinations: "syslog"
+        :services: ["pulpcore-content", "pulpcore-api", "pulpcore-workers@*"]
+        :debug:
+          -
+            :action: ensure_line_is_present
+            :line: ["            'level'", ": ", "'DEBUG',"]
```

Comment:
How does the code make sure that this line ends up in the right spot in the file without a declared anchor or descriptor of where it should exist?

Reply:
I tested this on a live instance with the real configuration generated by the installer (hacked a bit to get latest changes in pulpcore module, given that katello nightlies were broken until this morning)

I changed it back and forth multiple times between debug mode and production, validating by restarting services and observing appropriate verbosity in logs, as well as manually inspecting the config file each time

so I can state confidently that as the config file is currently structured, it works as expected. I am sure some corner case config file could be constructed to break it (given my intention to put https://github.com/theforeman/puppet-pulpcore/pull/210/files into the settings.py header I think I am OK with that though).

In particular, I copied the basic structure of this feature from the pulp2 equivalent, where in testing I noticed that it actually changes a commented line _describing_ log levels into the actual configuration (and also makes the change a few lines below where the setting already existed). This is admittedly janky but otherwise has no production impact... a commented line is destroyed and some helpful text is lost, replaced with a doubled config line. Once that change from the initial config file is made on the first run, both versions of the line continue to exist and both get updated on each subsequent log level change via hammer admin.

Reply:
One alternative that I considered is that the logging configuration could be a single line in settings.py :

`LOGGING = {"dynaconf_merge": True, "loggers": {'': {'handlers': ['console'], 'level': 'DEBUG'}}}`

I think there are other methods present here in hammer cli foreman admin to ensure the operation is more robust towards different things you may find in a hand edited config file, but I didn't explore it further as I was able to get it working reliably with the config file format we planned to ship, and we already are trying to warn users to never, ever make hand edits to that file (due to other non-obvious bugs they could easily introduce).

Reply:
P.S. I did author a "fix" for the destroyed comment / duplicate log level config issue in pulp2: https://github.com/theforeman/hammer-cli-foreman-admin/pull/13

:^)